pallets/committee/src/benchmarking.rs

```diff
@@ -0,0 +1,203 @@
+    }: {
+        call.dispatch_bypass_filter(origin)?;
+    }
+    verify {
+        ensure!(Module::<T, _>::vote_threshold() == (n, d), "incorrect vote threshold");
+    }
+
+    set_release_coordinator {
+        let dids: Vec<_> = (0..COMMITTEE_MEMBERS_NUM).map(|i| user::<T>("member", i).did()).collect();
+        let coordinator = dids.last().unwrap().clone();
+        Members::<I>::put(dids);
+        let origin = T::CommitteeOrigin::successful_origin();
+        let call = Call::<T, I>::set_release_coordinator(coordinator);
+    }: {
+        call.dispatch_bypass_filter(origin)?;
+    }
+    verify {
+        ensure!(
+            Module::<T, _>::release_coordinator() == Some(coordinator),
+            "incorrect release coordinator"
+        );
+    }
+
+    set_expires_after {
+        let maybe_block = MaybeBlock::Some(1.into());
+        let origin = T::CommitteeOrigin::successful_origin();
+        let call = Call::<T, I>::set_expires_after(maybe_block);
+    }: {
+        call.dispatch_bypass_filter(origin)?;
+    }
+    verify {
+        ensure!(Module::<T, _>::expires_after() == maybe_block, "incorrect expiration");
+    }
+
+    vote_or_propose_new_proposal {
+        let members = make_members_and_proposals::<T, I>()?;
+        let last_proposal_num = ProposalCount::<I>::get();
+        let (proposal, hash) = make_proposal::<T, I>(0);
+        identity::CurrentDid::put(members[0].did());
+    }: vote_or_propose(members[0].origin.clone(), true, Box::new(proposal.clone()))
+    verify {
+        // The proposal was stored.
+        ensure!(Proposals::<T, I>::get().contains(&hash), "new proposal hash not found");
+        ensure!(ProposalOf::<T, I>::get(&hash) == Some(proposal), "new proposal not found");
+    }
+
+    vote_or_propose_existing_proposal {
+        let members = make_members_and_proposals::<T, I>()?;
+        let (proposal, hash) = make_proposal::<T, I>(1);
+        let proposals = Proposals::<T, I>::get();
+        ensure!(proposals.contains(&hash), "cannot find the first proposal");
+        identity::CurrentDid::put(members[1].did());
+    }: vote_or_propose(members[1].origin.clone(), true, Box::new(proposal.clone()))
+    verify {
+        if COMMITTEE_MEMBERS_MAX <= 4 {
+            // Proposal was executed.
+            ensure!(
+                Module::<T, _>::voting(&hash).is_none(),
+                "votes are present on an executed existing proposal"
+            );
+        } else {
+            // The proposal was stored.
+            ensure!(Proposals::<T, I>::get().contains(&hash), "existing proposal hash not found");
+            ensure!(ProposalOf::<T, I>::get(&hash) == Some(proposal), "existing proposal not found");
+        }
+    }
+
+    vote {
+        // reject or approve
+        let a in 0 .. 1;
+
+        let members = make_members_and_proposals::<T, I>()?;
+        let hash = make_proposal::<T, I>(1).1;
+        let first_proposal_num = 0;
```

Comment:
If so, we might want to do this in a separate dedicated PR rather than in this one because of changes in behaviour.

Reply:
BTW the weight of the proposal is added to the result of the weight function. I think this should do until any changes in scheduling.

Reply:
@maxsam4 Would you still require scheduling changes in this PR or can we open a separate issue for that?

Reply:
It's not ideal because we don't account for the extra weight in the `vote` function. An attack vector will look something like this - 
1) Slowly build up a lot of proposals such that they only need one vote to execute.
2) Batch call vote to execute all of them in a single block. Since we do not account for the execution weight in `vote`, the system will accept plenty of `vote` calls in a single block however it will take a long time to process all of them. Thus, delaying block time.

That being said, this module can only be used by permission, committee members so this is an acceptable risk. Please add a low priority Jira task to track this. We do not need to fix this for v1 but it should be tracked nevertheless.